scalameta/scalameta/shared/src/test/scala/scala/meta/tests/parsers/XmlSuite.scala

```diff
@@ -236,6 +237,20 @@ class XmlSuite extends ParseSuite with DiffAssertions {
        |  </tr>
        |}""".stripMargin
   )
+  checkOK("<a>{<b>{1}</b>}</a>")
+  checkOK(
+    """val ips = <ips>{
+      |  for {
+      |    field <-ipsList
+      |    JString(ip) <- field.value
+      |  } yield <ip>{ ip }</ip>
+      |}</ips>
+    """.stripMargin)
+
+
+  //FIXME checkOK("<a>{}</a>")
```

Comment:
No, tokens should preserve all trivia, we must be able to distinguish between `{()}` and `{}` even if they have the same AST. I recommend we emit something like this (ignoring positions)
```
+      |                [5..5) ----> class scala.meta.tokens.Token$Xml$SpliceStart
 +      |              { [5..6) ----> class scala.meta.tokens.Token$LeftBrace
 +      |             } [9..10) ----> class scala.meta.tokens.Token$RightBrace
 +      |              [10..10) ----> class scala.meta.tokens.Token$Xml$SpliceEnd
```

and adapt the ScalametaParser to emit `Lit.Unit(())` if we encounter `LeftBrace` immediately followed by  `RightBrace` inside an xml splice. What do you think?